File: Ejercicio4.cpp
#include<iostream>
#include<fstream>
#include<sstream>
#include<string>
#include<iomanip>

using namespace std;

int main() {
	
	ifstream inputFile("C:\\LDP\\Temperaturas.txt");
	ofstream outputFile("registro de temperaturas.txt");
	
	if(!inputFile.is_open()) {
		cerr<<"No se pudo abrir el archivo de entrada "<<endl;
		return 1;
	}
	if(!outputFile.is_open()) {
		cerr<<"No se pudo abrir el archivo de salida "<<endl;
		return 1;
	}
	
	string line;
	string min, max;
	double maxTem = -250.0;
	double minTem = 200.0;
	
	while(getline(inputFile, line)) {
		istringstream iss(line);
		string fecha;
		double temperatura;
		
		getline(iss, fecha, ',');
		iss>>temperatura;
		
		
		if(temperatura > maxTem) {
			maxTem = temperatura;
			max = fecha;
		}
		if(temperatura < minTem) {
			minTem = temperatura;
			min = fecha;
		}
	}
	outputFile<<"Dia de temperatura maxima: "<<max<<", "<<maxTem<<endl;
	outputFile<<"Dia de temperatura minima: "<<min<<", "<<minTem<<endl;
	
	inputFile.close();
	outputFile.close();
	
	cout<<"El registro de temperaturas se guardo en 'registro de temperaturas'"<<endl;
	
    return 0; 
}
    

File: Ejercicio4.py
import streamlit as st

entrada = 'Temperaturas.txt'
t_max = -100
t_min = 99999
r_max = ''
r_min = ''
try:
    with open(entrada, 'r') as archivo:
        for line in archivo:
            partes = line.strip().split(', ')
            dia = partes[0]
            temperatura = float(partes[1])
            
            if temperatura > t_max:
                t_max = temperatura
                r_max = f'Dia de Temperatura maxima: {dia},{ temperatura}'
        
            if temperatura < t_min:
                t_min = temperatura
                r_min = f'Dia de Temperatura minima: {dia },{ temperatura}'
    st.title('Registro de Temperaturas')
    
    st.subheader('Registro:')
    st.write(r_max)
    st.write(r_min)

    salida = f'{r_max}\n{r_min}'

    st.download_button('Descargar Archivo', salida.encode('utf-8'), file_name='Registo de T.txt')
        
except FileNotFoundError:
    st.error(f'El archivo {entrada} no existe')